[Ryan Walter Lee] iP

ryanwalterlee opened the pull request:
# Duke Improved
> “Your mind is for having ideas, not holding them.” – David Allen [(source)](https://dansilvestre.com/productivity-quotes/)

It is able to:
1. Store all your tasks
2. Mark or Unmark as done
3. Filter based on keywords

The following are the commands that it contains:
- list
- todo
- deadline
- event
- mark
- unmark
- find

Coming Soon:
- [x] Filter based on keywords
- [ ] Error Handling
- [ ] list of commands

If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```
public static void main(String[] args) throws DukeException, IOException {
        Duke duke = new Duke(Storage.readFile());
        duke.run();
}
```

Best of all, its **FREE** :grinning:

Commit:
Add Gradle support

ryanwalterlee committed:
Level-1

ryanwalterlee committed:
Level-2

ryanwalterlee committed:
Level-3

ryanwalterlee committed:
Level-4 add todo, deadline, events

ryanwalterlee committed:
A-TextUiTesting

ryanwalterlee committed:
Level-5 error handling

ryanwalterlee committed:
Level-6: Delete

ryanwalterlee committed:
A-Enums: Use Enums

ryanwalterlee committed:
add save task to hard disk functionality

ryanwalterlee committed:
change storage of string date to localdate

ryanwalterlee committed:
Merge branch 'branch-Level-7'

ryanwalterlee committed:
change event to also include localdate

ryanwalterlee committed:
Merge branch 'branch-level-8'

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Event.java

ryanwalterlee committed:
abstract storage and parser

ryanwalterlee committed:
abstract ui and tasklist

ryanwalterlee committed:
Add into package duke

ryanwalterlee committed:
Add Junit Tests

ryanwalterlee committed:
Configure to cater to jar file

ryanwalterlee committed:
Add java doc comments

ryanwalterlee committed:
Change to follow coding standard

ryanwalterlee committed:
Add find task command

ryanwalterlee committed:
Merge branch 'branch-A-JavaDoc'

ryanwalterlee committed:
Update gitignore

ryanwalterlee committed:
Update data file

ryanwalterlee committed:
Merge branch 'branch-A-CodingStandard'

ryanwalterlee committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/Parser.java

ryanwalterlee commented:
# Duke Improved
> “Your mind is for having ideas, not holding them.” – David Allen [(source)](https://dansilvestre.com/productivity-quotes/)

It is able to:
1. Store all your tasks
2. Mark or Unmark as done
3. Filter based on keywords

The following are the commands that it contains:
- list
- todo
- deadline
- event
- mark
- unmark
- find

Coming Soon:
- [x] Filter based on keywords
- [ ] Error Handling
- [ ] list of commands

If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```
public static void main(String[] args) throws DukeException, IOException {
        Duke duke = new Duke(Storage.readFile());
        duke.run();
}
```

Best of all, its **FREE** :grinning:

ryanwalterlee committed:
Merge branch 'add-gradle-support'

ryanwalterlee committed:
Do Gui up to part 2

ryanwalterlee committed:
Complete Gui

ryanwalterlee committed:
Complete GUI

ryanwalterlee committed:
Merge branch 'branch-Level-10'

kxshxsh commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,37 @@
+        this.deadline = LocalDate.parse(deadline);
+    }
+
+    private String dateToString() {
```
The java coding standard says that "Names representing methods must be verbs and written in camelCase." dateToString() in Deadline.java and Event.java don't fit that in my opinion. Maybe something you'd like to consider and change to be more in line with the standard?

ryanwalterlee replied:
You're right, how about convertDateToString?

kxshxsh commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,51 @@
+package duke;
+
+import java.io.IOException;
```
I appreciate the way that you have ordered your import statements, they are easy to comprehend and stick to the standard's specification about grouping:)

kxshxsh commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,121 @@
+    /**
+     * deletes task at a specified position and prints a confirmation
+     *
+     * @param number represents which task to delete, 1 for the first task, 2 for the second, etc
```
I think you have done a thorough job in writing your javadocs; however, just nitpicking here to bring minute details to your attention:) 
"Punctuation behind each parameter description". In that sense, there would be a period after 'etc' and similarly for other javadocs:))

ryanwalterlee replied:
Yup, it appears I have missed out on all the full stops after each line, I will fix this issue, thanks for spotting.

kxshxsh reviewed (commented):
Hi! I admire the effort you've put into your project! I think it's very well written and overall- hardly strays away from coding standards! Major kudos to you:) I have highlighted the tiniest errors that I have noticed that could help improve its quality

s7manth commented on src/main/java/duke/DukeException.java:
```diff
@@ -0,0 +1,14 @@
+/**
+ * Custom Exception for Duke related Errors
+ */
+public class DukeException extends Exception {
```
Extend from the `RunTimeException`. 

`Exception` is more broad and can leave out the granularities. Always remember to catch the exception with the smallest scope.

s7manth commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,37 @@
+
+    protected LocalDate deadline;
+
+    Deadline(String description, String deadline) {
```
Constructors here are initialized with the `default` access modifier. `default` makes them package-private. You could have made them `public` for accessing across packages, if needed.

s7manth commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,37 @@
+        return time.format(DateTimeFormatter.ofPattern("MMM dd yyyy"));
+    }
+
+    String getTaskType() {
```
Access modifiers are missing.

s7manth commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,121 @@
+    }
+
+    /**
+     * adds the given Event object and prints a confirmation
```
Java Doc comments must start with a capital letter and end with a full stop according to 2103T's convention.

s7manth commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,30 @@
+        this.tasks = new ArrayList<>();
+    }
+
+    int size() {
```
Java Doc comments are missing for a few classes.

s7manth commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,134 @@
+        this.tasks = tasks;
+    }
+
+    enum Commands {
+        LIST("list"),
+        MARK("mark"),
+        UNMARK("unmark"),
+        TODO("todo"),
+        DEADLINE("deadline"),
+        EVENT("event"),
+        DELETE("delete"),
+        FIND("find");
+
```
Cool usage of enums.

brelkh replied:
The way you used enums was appropriate and relevant to the project. Additionally, you followed the coding standard and used PascalCase to name this enum rather than camelCase. 👍

s7manth reviewed (commented):
Good quality code. Just a few suggestions here and there. Nothing serious. LGTM

brelkh commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,37 @@
+
+public class Event extends Task {
+
+    protected LocalDate time;
+
+    Event(String description, String time) {
+        super(description);
+        this.time = LocalDate.parse(time);
+    }
```
While `time` is an appropriate attribute name, the textbook mentions that a name could explain the named entity accurately and to a sufficient level of detail. Perhaps a name like `timeOfEvent` or `eventTime` would be more clear to readers?

brelkh commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,121 @@
+
+
+    /**
+     * marks the chosen task as done and prints a confirmation
+     *
+     * @param taskNumber represents which task to mark, 1 for the first task, 2 for the second, etc
+     */
```
Adding the `@return` tag to public javadoc comments could help readers better understand the method, and, as an added bonus, more useful information will be shared by IDEs in their quick documentation feature.

brelkh commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,121 @@
+        this.tasks = tasks;
+    }
+
+    /**
+     * prints the list of currently stored tasks
+     */
```
As shown in the java coding standard guide, sentences in javadoc comments usually begin with a capital letter and end with a full-stop. It's definitely not a major error, but adhering closely to the coding standard could help achieve the overall goal of making your future tP code look like it was all written by a single person.

brelkh reviewed (commented):
Your code LGTM overall! However, ensuring your javadoc comments follow the coding standard could improve code readability and make the comments clearer for readers to understand. :)

ryanwalterlee committed:
Edit Javadocs to follow coding standard and configure build.gradle

ryanwalterlee committed:
Update code with assertions where needed

Assertions can help to document important assumptions that should hold at various points of the code.

Assertions are placed especially where the ui parses the command, if the command is not correct, then the assertions will catch them.

Assertions can help during development phase to check if the assumptions hold true, this improves the speed of debugging.

ryanwalterlee committed:
Improve Code quality based on rules stated in CS2103 website

Code quality is important because it helps improve readability of
code. This way, if other people were to be working on the same code,
they would have an easier time understanding the code. There is also
the benefit of the author himself understanding the code better.

For each of the files, rules based on the CS2103 website are
strictly adhered to to ensure readability of code.

Code quality is updated now as we have accumulated some technical
debt and it is about time to pay it.

ryanwalterlee committed:
Merge pull request #1 from ryanwalterlee/branch-Assertions

Update code with assertions where needed

ryanwalterlee committed:
Merge branch 'master' of https://github.com/ryanwalterlee/ip into branch-CodeQuality

ryanwalterlee committed:
Merge pull request #2 from ryanwalterlee/branch-CodeQuality

Improve code quality

ryanwalterlee committed:
Add help command

ryanwalterlee committed:
Merge branch 'branch-C-Help'

ryanwalterlee committed:
Add user guide

ryanwalterlee committed:
Fix Readme formatting

ryanwalterlee committed:
Update Readme to include how to run jar

ryanwalterlee committed:
Fix readme formatting for header

ryanwalterlee committed:
Fix delete bug
